internal/app/machined/pkg/controllers/network/dns_resolve_cache.go

```diff
@@ -288,3 +297,49 @@ func newDNSRunner(cfg runnerConfig, cache *dns.Cache, logger *zap.Logger) (*dns.
 
 	return dns.NewServer(serverOpts), nil
 }
+
+type stateMapper struct {
+	state state.State
+}
+
+func (s *stateMapper) ResolveAddr(ctx context.Context, qType uint16, name string) []netip.Addr {
+	name = strings.TrimRight(name, ".")
+
+	list, err := safe.ReaderListAll[*cluster.Member](ctx, s.state)
+	if err != nil {
+		return nil
+	}
+
+	elem, ok := list.Find(func(res *cluster.Member) bool {
+		return fqdnMatch(name, res.TypedSpec().Hostname) || fqdnMatch(name, res.TypedSpec().NodeID)
```

Comment:
`Nodename`, not `NopeID`

Reply:
There is no `Nodename` in the member spec, but `AffiliateSpec` has it. But it is used to populate the id of the member resource [here](https://github.com/siderolabs/talos/blob/fa3b93370501009283e110b74876b18ce6bad4f9/internal/app/machined/pkg/controllers/cluster/member.go#L75) so I think we can use that.

Reply:
ah yeah, right, it's the Member